pkg/gcv/cf/cf_test.go

```diff
@@ -955,6 +955,81 @@ spec:
 	}
 }
 
+func TestDefaultMatcher(t *testing.T) {
+	testCases := []struct {
+		description  string
+		target       []string
+		exclude      []string
+		ancestryPath string
+		wantMatch    bool
+	}{
+		{
+			description:  "default matches org",
+			ancestryPath: "organization/1",
+			wantMatch:    true,
+		},
+		{
+			description:  "default matches folder",
+			ancestryPath: "organization/1/folder/2",
+			wantMatch:    true,
+		},
+		{
+			description:  "default matches project",
+			ancestryPath: "organization/1/folder/2/project/3",
+			wantMatch:    true,
+		},
+		{
+			description:  "exclude works with default",
+			exclude:      []string{"organization/1/folder/2"},
+			ancestryPath: "organization/1/folder/2/project/3",
+			wantMatch:    false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			cf, err := New(map[string]string{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = cf.AddTemplate(makeTestTemplate("template"))
```

Comment:
nit: make a new function for an always failing constraint template as a new function
eg: 
func makeAlwaysFailTemplate()

Reply:
This template is like any other test template; it's the constraint that's triggering the failures.
I thought of extracting a method to create the always failing constraint but decided against it because there are 2 different versions (one with target field and the other without) and so far there's no duplication yet.